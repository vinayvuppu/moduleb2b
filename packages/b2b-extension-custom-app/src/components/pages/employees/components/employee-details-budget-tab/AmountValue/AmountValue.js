import React from 'react';
import PropTypes from 'prop-types';
import { formatMoney } from '@commercetools-local/utils/formats/money';
import { useIntl } from 'react-intl';
import { Text, Spacings } from '@commercetools-frontend/ui-kit';

const AmountValue = ({ amount, label, noValueLabel, errorOnNegative }) => {
  const intl = useIntl();

  const getValue = () => {
    if (!amount) return noValueLabel;
    if (amount.centAmount <= 0 && errorOnNegative)
      return <span style={{ color: 'red' }}>{formatMoney(amount, intl)}</span>;

    return formatMoney(amount, intl);
  };

  return (
    <Spacings.Stack>
      <Text.Body isBold>{label}</Text.Body>
      <Text.Headline as="h3" isBold>
        {getValue()}
      </Text.Headline>
    </Spacings.Stack>
  );
};

AmountValue.propTypes = {
  amount: PropTypes.shape({
    centAmount: PropTypes.number.isRequired,
  }),
  label: PropTypes.string.isRequired,
  noValueLabel: PropTypes.string.isRequired,
  errorOnNegative: PropTypes.bool,
};

AmountValue.defaultProps = {
  errorOnZero: false,
};

AmountValue.displayName = 'AmountValue';

export default AmountValue;
