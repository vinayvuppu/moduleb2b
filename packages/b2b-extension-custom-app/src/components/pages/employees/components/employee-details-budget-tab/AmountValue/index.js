import AmountValue from './AmountValue';

export default AmountValue;
