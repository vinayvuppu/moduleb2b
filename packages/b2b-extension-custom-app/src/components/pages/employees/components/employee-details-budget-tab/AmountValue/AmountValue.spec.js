import React from 'react';
import { render, cleanup } from '@testing-library/react';
import AmountValue from './AmountValue';

jest.mock('react-intl', () => {
  const actual = jest.requireActual('react-intl');
  return {
    ...actual,
    useIntl: jest.fn(() => ({
      formatNumber: jest.fn(),
      formatMessage: jest.fn(() => 'formatted message'),
    })),
  };
});

afterEach(cleanup);

describe('AmountValue component', () => {
  describe('rendering', () => {
    it('should render the correct snapshot', () => {
      const { asFragment } = render(
        <AmountValue label={'foo'} noValueLabel={'foo-no-value'} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
