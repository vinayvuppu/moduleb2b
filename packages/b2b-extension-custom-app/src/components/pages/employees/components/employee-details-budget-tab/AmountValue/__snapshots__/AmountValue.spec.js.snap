// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountValue component rendering should render the correct snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-z1wx52-getStyles"
  >
    <p
      class="css-17wfq1-bodyStyles"
    >
      foo
    </p>
    <h3
      class="css-19gfu0y-headlineStyles"
    >
      foo-no-value
    </h3>
  </div>
</DocumentFragment>
`;
